reactor/reactor-netty
# Migrate 1.2.x doc to Antora

This PR is first attempt to migrate Reactor Netty 1.2.x documentation to Antora 

There are three logical commits in this PR:

- bfb304d6e01e76a5765efcc5271e3ca8600ffbef: it contains all modifs done in .adoc files, and also contains antora yaml files
-  cf9365f166b596fee3d4fefdcaec9c02b7267bcf: contains all adaptation done in various gradle scripts
- d3bf58953487aa196054461eb7cab3ad738aac19: contains adaptations done in the GH worfklows.


To test this PR, you can proceed like the following:

### Testing without PDF:

- clone this PR
- switch to java 17 (the docs sub project requires java17)
- build the doc and generate it to docs/build/site/index.html:
```
./gradlew antora
```
- you can now browse docs/build/site/index.html
- switch to java 8
- deploy reactor-netty docs to your local M2 
```
./gradlew publishToMavenLocal
```
- now, you can verify the content of ~/.m2/repository/io/projectreactor/netty/reactor-netty/1.2.0-SNAPSHOT/reactor-netty-1.2.0-SNAPSHOT-docs.zip or  ~/.m2/repository/io/projectreactor/netty/reactor-netty/1.2.0-SNAPSHOT/reactor-netty-1.2.0-SNAPSHOT-javadoc.jar

### Testing with PDF:

As before, PDF are not generated by default in snapshot mode. If you want to also include a PDF in the generated docs zip, then do the following:

- first, you will need to install **asciidoctor-pdf** command, because the antora pdf assembler requires it (this command is provided when doing "brew install asciidoctor"):
```
brew install asciidoctor
```
- switch to java17
- build antora and force PDF generation (in release mode, no need to use -PforcePdf, it's like before):
```
./gradlew antora -PforcePdf
this will generate  ./docs/build/assembler/reactor-netty/1.2.0/reactor-netty-reference-guide.pdf
```
- switch to java 8
- deploy reactor-netty docs to your local M2:
```
./gradlew publishToMavenLocal -PforcePdf
```
- verify the content of ~/.m2/repository/io/projectreactor/netty/reactor-netty/1.2.0-SNAPSHOT/reactor-netty-1.2.0-SNAPSHOT-docs.zip, which should also include a pdf in the root dir of the zip.
